esmvaltool/diag_scripts/hydrology/lisflood.py

```diff
@@ -138,14 +141,20 @@ def main(cfg):
         cubes['pr'].units = 'mm d-1'
 
         for var_name, cube in cubes.items():
-            cube.remove_coord('shape_id')
             # Western emisphere longitudes should be negative
             points = cube.coord('longitude').points
             cube.coord('longitude').points = (points + 180) % 360 - 180
             # latitudes decreasing
             cube = cube[:, ::-1, ...]
 
-            output_file = save(cube, var_name, dataset, cfg)
+            # convert to xarray dataset (xrds)
+            # remove coordinate bounds drop extra coordinates and reorder
+            xrds = xr.DataArray.from_iris(cube).to_dataset(promote_attrs=True)
```

Comment:
```suggestion
            xrds = xr.DataArray.from_iris(cube).to_dataset()
```
`promote_attrs` has only recently been added, and seems not needed to preserve the functionality.

Reply:
@Peter9192 I thought it was necessary, because I lost all metadata when I added this "to_dataset()". Maybe I was mistaken?

Reply:
@JaroCamphuijsen Well, the difference is that an `xarray.dataset` is a collection of `xarray.dataarray`'s, and if you look at the individual dataarray again (xrds.tas for example), they still have the correct metadata.

Reply:
Alright! I though I checked that after saving I didn't have the metadata in the file. However now I think about it this seems unlikely and probably I just didn't check thoroughly.